### PR TITLE
Refactor tests to include DisplayNames and better method names

### LINKED ISSUE #411: Refactor tests to include DisplayNames for classes and better method names
In line with our recent change to our style guide, [all classes should be annotated with a display name](https://github.com/Project-Books/book-project/blob/master/STYLEGUIDE.md#using-displayname).

Implement this change. For example, `AuthorTest` would have the display name `@DisplayName("Author should")`.

In addition to adding a display name, refactor the method names so they read nicer as a sentence. For example, in `PredefinedShelfUtilsTest`, the method `testGetPredefinedShelfNamesAsStrings()` could become `returnsCorrectPredefinedShelfNames()`. This way, the 'sentence' would be "PredefinedShelfUtilsTest should returnCorrectPredefinedShelfNames()`.

### PR DESCRIPTION
**Pull request by schrothbn**
## Summary of change

Refactored all tests (if necessary) to include DisplayNames for classes and better method names

## Related issue

Closes #411 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ)

### CODE REVIEW

**Comment by knjk04** [comment]
@schrothbn Thanks again for the quick turnaround from being assigned to creating the PR! We'll review this shortly

File: src/test/java/com/karankumar/bookproject/backend/entity/account/UserTest.java [hunk 1]
```diff
@@ -41,7 +43,7 @@ void setUp() {
     }
 
     @Test
-    void testValidUserSaved() {
+    void beSaved() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/entity/account/UserTest.java:46]: How about adding `ifValid` to the end of this?

File: src/test/java/com/karankumar/bookproject/backend/goal/CalculateReadingGoalTest.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ static void setUp() {
     }
 
     @Test
-    void testProgressValueIsCorrect() {
+    void calculateProgressValueCorrect() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/goal/CalculateReadingGoalTest.java:44]: How about changing this to `calculateCorrectProgressValue`?

File: src/test/java/com/karankumar/bookproject/backend/goal/CalculateReadingGoalTest.java [hunk 3]
```diff
@@ -50,14 +50,14 @@ void testProgressValueIsCorrect() {
     }
 
     @Test
-    void testProgressValueIsCorrectWhenGoalMet() {
+    void calculateProgressValueCorrectWhenGoaldMet() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/goal/CalculateReadingGoalTest.java:53]: What about `calculateCorrectProgressValueWhenGoalMet`?

File: src/test/java/com/karankumar/bookproject/backend/goal/CalculateReadingGoalTest.java [hunk 4]
```diff
@@ -50,14 +50,14 @@ void testProgressValueIsCorrect() {
...
         double expected = 1.0;
         double actual = calculateProgressTowardsReadingGoal(BOOKS_TO_READ, BOOKS_TO_READ);
         assertThat(actual).isEqualTo(expected);
     }
 
     @Test
-    void testProgressIsCorrectWhenGoalExceeded() {
+    void calculateProgressValueCorrectWhenGOalExceeded() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/goal/CalculateReadingGoalTest.java:60]: The 'O' in 'Goal' should be lowercase

File: src/test/java/com/karankumar/bookproject/backend/repository/CustomShelfRepositoryTest.java [hunk 5]
```diff
@@ -28,12 +28,12 @@
...
 
 @DataJpaIntegrationTest
+@DisplayName("CustomShelfRepository should")
 class CustomShelfRepositoryTest {
     @Autowired private CustomShelfRepository repository;
     private final String customShelf1 = "Test1";
 
     @Test
-    @DisplayName("When shelf exists, findByShelfName returns one shelf")
     void findByShelfNameCorrectlyReturnsOneShelf() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/repository/CustomShelfRepositoryTest.java:37]: Could you please rename to 'findCorrectShelf'?

File: src/test/java/com/karankumar/bookproject/backend/repository/PredefinedShelfRepositoryTest.java [hunk 6]
```diff
@@ -49,8 +50,7 @@ void setup() {
     }
 
     @Test
-    @DisplayName("When a shelf exists, findByPredefinedShelfName correctly returns the shelf")
-    void findByShelfNameReturnsOneShelf() {
+    void correctlyReturnExistingShelf() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/repository/PredefinedShelfRepositoryTest.java:53]: Could you please rename this to something like 'findCorrectShelf'?

File: src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java [hunk 7]
```diff
@@ -83,13 +84,13 @@ private void resetServices() {
...
     @Test
-    void nullBookNotSaved() {
+    void notSaveNullBook() {
         bookService.save(null);
         assertThat(bookService.count()).isZero();
     }
 
     @Test
-    void bookWithoutAuthorNotSaved() {
+    void notSaveBookWithout() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java:93]: I think you forgot 'Author' at the end of this?

File: src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java [hunk 8]
```diff
@@ -92,70 +94,70 @@ void register_withEmailTaken_throwsException() {
...
     @Test
-    void register_withoutUserRole_throwsError() {
+    void throwExceptionOnRegisterWithoutUserRole() {
         assertThatThrownBy(() -> userService.register(validUser))
                 .isInstanceOf(AuthenticationServiceException.class);
     }
 
     @Test
-    void register_registersUser() {
+    void registerUser() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java:103]: How about `registerValidUser` (I know the name before didn't have this, but I think this is clearer)?

File: src/test/java/com/karankumar/bookproject/backend/statistics/GenreStatisticsTest.java [hunk 9]
```diff
@@ -42,21 +44,21 @@ public static void setup(@Autowired BookService bookService,
     }
 
     @Test
-    void mostReadGenreExistsAndIsFound() {
+    void findExistingMostReadGenre() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/statistics/GenreStatisticsTest.java:47]: What about `findMostReadGenre()`?

File: src/test/java/com/karankumar/bookproject/backend/statistics/GenreStatisticsTest.java [hunk 10]
```diff
@@ -42,21 +44,21 @@ public static void setup(@Autowired BookService bookService,
...
         BookGenre expected = StatisticTestUtils.MOST_READ_BOOK_GENRE;
         BookGenre actual = genreStatistics.findMostReadGenre();
         assertThat(actual).isEqualTo(expected);
     }
 
     @Test
-    void mostLikedGenreExistsAndIsFound() {
+    void findMostLikedGenreExists() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/statistics/GenreStatisticsTest.java:54]: What about `findMostLikedGenre()`?

File: src/test/java/com/karankumar/bookproject/backend/statistics/PageStatisticsTest.java [hunk 11]
```diff
@@ -52,13 +54,13 @@ public void beforeEachSetup() {
...
     @Test
-    void bookWithMostPagesExistsAndIsFound() {
+    void findBookWithMostPages() {
         assertThat(pageStatistics.findBookWithMostPages().getTitle())
                 .isEqualTo(StatisticTestUtils.getBookWithMostPages().getTitle());
     }
 
     @Test
-    void onlyReadBooksCountTowardsMostPagesStatistics() {
+    void coundOnlyReadBooksTowardsMostPagesStatistics() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/statistics/PageStatisticsTest.java:63]: Typo here: this should be 'count' at the start

File: src/test/java/com/karankumar/bookproject/backend/statistics/RatingStatisticsTest.java [hunk 12]
```diff
@@ -55,41 +57,41 @@ public void beforeEachSetup() {
     }
 
     @Test
-    void lowestRatedBookExistsAndIsFound() {
+    void findLowestRatedBookExists() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/statistics/RatingStatisticsTest.java:60]: Could you please remove 'Exists'?

File: src/test/java/com/karankumar/bookproject/backend/utils/CustomShelfUtilsTest.java [hunk 13]
```diff
@@ -82,20 +84,20 @@ private static void saveCustomShelves(CustomShelf... customShelves) {
...
-    void givenNoBooksInCustomShelf_getBooksInCustomShelfReturnsNoBooks() {
+    void returnNoBooksIfNoBooksInCustomShelf() {
         Set<Book> actual =
                 customShelfUtils.getBooksInCustomShelf(customShelfWithNoBooks.getShelfName());
         assertThat(actual).isEmpty();
     }
 
     @Test
-    void givenInvalidCustomShelf_anEmptySetOfBooksAreReturned() {
+    void returnEmptySetOfBooksWhenCustomShelfIsInvalid() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/utils/CustomShelfUtilsTest.java:100]: Could you please rename this to 'returnEmptySetofBooksFromNonExistentCustomShelf'?

File: src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java [hunk 14]
```diff
@@ -347,7 +347,7 @@ void correctFormFieldsShowForDidNotFinishShelf() {
 
     @ParameterizedTest
     @MethodSource("shelfNames")
-    void fieldsToResetAreCorrectlyPopulated(
+    void populateCorrectlyFieldsToReset(
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java:350]: Could you please move 'Correctly' to the end (so we get 'populateFieldsToResetCorrectly')?

File: src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java [hunk 15]
```diff
@@ -404,7 +404,7 @@ void shouldNotAllowNegativePageNumbers() {
     }
 
     @Test
-    void testNumberOfPagesFieldShouldNotExceedMaxValue() {
+    void notAllowToExceedMaxValueNumberOfPagesField() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java:407]: Could you please rename to 'notAllowPagesFieldToExceedMaxValue'?

File: src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java [hunk 16]
```diff
@@ -416,7 +416,7 @@ void testNumberOfPagesFieldShouldNotExceedMaxValue() {
     }
 
     @Test
-    void testPagesReadShouldNotExceedMaxValue() {
+    void notAllowToExceedMaxValueToPagesRead() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java:419]: Could you please rename to 'notAllowPagesReadToExceedMaxValue'?

File: src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java [hunk 17]
```diff
@@ -452,7 +452,7 @@ void shouldNotAllowEmptyAuthorFirstName() {
     }
 
     @Test
-    void changeAuthorNameShouldBeSaved() {
+    void shouldSaveChangedAuthorName() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java:455]: As we already have 'should' in the class's display name, could you please rename this to 'saveChangedAuthorName'?

File: src/test/java/com/karankumar/bookproject/ui/registration/RegistrationFormTest.java [hunk 18]
```diff
@@ -92,7 +94,7 @@ void passwordFieldNotMatchingTheRulesHasError() {
     }
 
     @Test
-    void passwordConfirmationWrongHasError() {
+    void setErrorOnPasswordConfirmationWrong() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/registration/RegistrationFormTest.java:97]: Could you please rename this to 'showErrorWhenPasswordConfirmationIsWrong'?

File: src/test/java/com/karankumar/bookproject/ui/registration/RegistrationFormTest.java [hunk 19]
```diff
@@ -83,7 +85,7 @@ public void setup() {
     }
 
     @Test
-    void passwordFieldNotMatchingTheRulesHasError() {
+    void setErrorOnPasswordFieldNotMatchingTheRules() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/registration/RegistrationFormTest.java:88]: Could you please change the 'setError' prefix to 'showError'?

File: src/test/java/com/karankumar/bookproject/ui/shelf/CustomShelfFormTest.java [hunk 20]
```diff
@@ -83,7 +85,7 @@ private void formIsInInitialState() {
     }
 
     @Test
-    void saveButtonShouldBeGreyedOutWithExistingShelfName () {
+    void disableSaveButtonInitiallyWithExistingShelfName () {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/shelf/CustomShelfFormTest.java:88]: Could you please remove 'Initially' here?

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks a lot for doing this! I've left some review feedback

File: src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java [hunk 21]
```diff
@@ -92,70 +94,70 @@ void register_withEmailTaken_throwsException() {
...
         roleRepository.save(Role.builder().role("USER").build());
         userService.register(validUser);
 
         assertThat(userRepository.findByUsername(validUser.getUsername())).isPresent();
     }
 
     @Test
-    void register_logsUserIn() {
+    void logsUserInAfterRegister() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java:111]: Could you please remove the 's' from 'logs'?

File: src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java [hunk 22]
```diff
@@ -92,70 +94,70 @@ void register_withEmailTaken_throwsException() {
...
         roleRepository.save(Role.builder().role("USER").build());
         userService.register(validUser);
 
         assertThat(SecurityContextHolder.getContext().getAuthentication().isAuthenticated()).isTrue();
     }
 
     @Test
-    void usernameIsInUse_UsernameNotInUse_returnsFalse() {
+    void checksIfUserNameIsInUseWithUsernameNotInUse() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java:119]: How about 'correctlyReportUsernameIsNotInUse'?

File: src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java [hunk 23]
```diff
@@ -92,70 +94,70 @@ void register_withEmailTaken_throwsException() {
...
         assertThat(userService.usernameIsInUse("notAtestuser")).isFalse();
     }
 
     @Test
-    void usernameIsInUse_UsernameInUse_returnsTrue() {
+    void checksIfUserNameIsInUseWithUsernameInUse() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java:124]: How about 'correctlyReportUsernameIsInUse'?

File: src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java [hunk 24]
```diff
@@ -92,70 +94,70 @@ void register_withEmailTaken_throwsException() {
...
-    void usernameIsNotInUse_UsernameInUse_returnsFalse() {
+    void checksIfUserNameIsNotInUseWithUsernameInUse() {
         userRepository.save(validUser);
 
         assertThat(userService.usernameIsNotInUse(validUser.getUsername())).isFalse();
     }
 
     @Test
-    void emailIsInUse_EmailNotInUse_returnsFalse() {
+    void checkIfEmailIsInUseWithEmailNotInUse() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java:143]: How about 'correctlyReportEmailIsNotInUse'?

File: src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java [hunk 25]
```diff
@@ -92,70 +94,70 @@ void register_withEmailTaken_throwsException() {
...
         assertThat(userService.emailIsInUse("testmail")).isFalse();
     }
 
     @Test
-    void emailIsInUse_EmailInUse_returnsTrue() {
+    void checkIfEmailIsInUseWithEmailInUse() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java:148]: How about 'correctlyReportEmailIsInUse'?

**Review by knjk04 — CHANGES_REQUESTED** [review]
A few more comments

**Comment by knjk04** [comment]
Also, could you please resolve the merge conflicts? I know there are lots, so I'd like to merge this PR in as soon as possible

**Comment by schrothbn** [comment]
> Also, could you please resolve the merge conflicts? I know there are lots, so I'd like to merge this PR in as soon as possible.
> 
> When you're ready for another review, let us know

Of course i'm on it. I didn't notice before my push, sorry.

**Review by knjk04 — APPROVED** [review]
Thanks for working on this! I appreciate there were lots of files and changes to make!

**Pull request merged**